starknet_py/tests/e2e/client/gateway_test.py

```diff
@@ -47,3 +50,53 @@ async def test_get_transaction_status(invoke_transaction_hash, gateway_client):
     assert isinstance(tx_status_resp, TransactionStatusResponse)
     assert tx_status_resp.transaction_status == TransactionStatus.ACCEPTED_ON_L2
     assert isinstance(tx_status_resp.block_hash, int)
+
+
+# pylint: disable=protected-access
+@pytest.mark.parametrize(
+    "net, chain",
+    ((TESTNET, StarknetChainId.TESTNET), (MAINNET, StarknetChainId.MAINNET)),
+)
+def test_creating_client_from_predefined_network(net, chain):
+    gateway_client = GatewayClient(net=net)
+    net_address = net_address_from_net(net)
+
+    assert gateway_client.net == net
+    assert gateway_client._feeder_gateway_client.url == f"{net_address}/feeder_gateway"
+    assert gateway_client._gateway_client.url == f"{net_address}/gateway"
+    assert gateway_client.chain == chain
+
+
+def test_creating_client_with_custom_net(run_devnet):
+    gateway_client = GatewayClient(net=run_devnet, chain=StarknetChainId.TESTNET)
+
+    assert gateway_client.net == run_devnet
+    assert gateway_client._feeder_gateway_client.url == f"{run_devnet}/feeder_gateway"
+    assert gateway_client._gateway_client.url == f"{run_devnet}/gateway"
+    assert gateway_client.chain == StarknetChainId.TESTNET
+
+
+def test_creating_client_with_custom_net_dict(run_devnet):
+    net = {
+        "feeder_gateway_url": f"{run_devnet}/feeder_gateway",
+        "gateway_url": f"{run_devnet}/gateway",
+    }
+
+    gateway_client = GatewayClient(net=net, chain=StarknetChainId.TESTNET)
+
+    assert gateway_client.net == net
+    assert gateway_client._feeder_gateway_client.url == net["feeder_gateway_url"]
+    assert gateway_client._gateway_client.url == net["gateway_url"]
+    assert gateway_client.chain == StarknetChainId.TESTNET
+
+
+def test_throwing_on_custom_net_dict_without_chain(run_devnet):
```

Comment:
What if we used a custom net but as a string and not a dict. Perhaps we should check if en error is also raised in that case.